src/app/ngrx/effects/castle.effects.ts

```diff
@@ -1,54 +1,231 @@
+      }));
+    })
+  );
+
+  // on podcast page success if loading all podcast pages and not yet finished, load next page
+  @Effect({dispatch: false})
+  loadNextPodcastPage$: Observable<Action> = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_PAGE_SUCCESS),
+    filter((action: ACTIONS.CastlePodcastPageSuccessAction) => {
+      const { page, all, total } = action.payload;
+      return all && page * PODCAST_PAGE_SIZE < total;
+    }),
+    map((action: ACTIONS.CastlePodcastPageSuccessAction) => action.payload),
+    concatMap((payload: ACTIONS.CastlePodcastPageSuccessPayload) => {
+      const { page, all } = payload;
+      return Observable.of(new ACTIONS.CastlePodcastPageLoadAction({page: page + 1, all}));
+    })
+  );
+
+  /*
+  // on podcast page 1 success, if loading all podcasts start loading all episodes page by page as well
+  // ugh, makes app loading too slow
+  @Effect()
+  loadPodcastEpisodes$: Observable<Action> = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_PODCAST_PAGE_SUCCESS),
+    filter((action: ACTIONS.CastlePodcastPageSuccessAction) => {
+      const { all, page, podcasts } = action.payload;
+      return all && page === 1 && podcasts && podcasts.length > 0;
+    }),
+    map((action: ACTIONS.CastlePodcastPageSuccessAction) => action.payload.podcasts),
+    mergeMap((podcasts: Podcast[]) => {
+      return podcasts.map(podcast => new ACTIONS.CastleEpisodePageLoadAction(
+        {podcastId: podcast.id, page: 1, all: true}));
+    })
+  );*/
+
+  // basic - load > success/failure episode page
+  @Effect()
+  loadEpisodePage$: Observable<Action> = this.actions$.pipe(
+    ofType(ACTIONS.ActionTypes.CASTLE_EPISODE_PAGE_LOAD),
+    map((action: ACTIONS.CastleEpisodePageLoadAction) => action.payload),
+    concatMap((payload: ACTIONS.CastleEpisodePageLoadPayload) => {
+      const { podcastId, page, all } = payload;
+      return this.castle.follow('prx:podcast', {id: podcastId}).followItems('prx:episodes', {page, per: EPISODE_PAGE_SIZE})
+        .pipe(
+          map((results: HalDoc[]) => {
+            return new ACTIONS.CastleEpisodePageSuccessAction({
+              page,
+              all,
+              total: results[0].total(),
+              episodes: results.map(doc => {
+                return {
+                  guid: '' + doc['id'],
+                  podcastId,
+                  publishedAt: doc['publishedAt'] ? new Date(doc['publishedAt']) : null,
+                  title: doc['title'],
+                  page
+                };
+              })
+            });
+          }),
+          catchError(error => Observable.of(new ACTIONS.CastleEpisodePageFailureAction({error})))
+        );
+    })
+  );
+
+  // on episode page success if loading all episode pages and not yet finished, load next page
+  @Effect()
+  loadNextEpisodePage$: Observable<Action> = this.actions$.pipe(
```

Comment:
I don't _need_ to load all the pages of episodes on app load just yet, but it's coming soon... On a podcast with a lot of episodes, the pages continue to load for a bit. I could definitely increase the page size, but the page size is kind of tangled up in the table paging as well, so it was really starting to seem out of scope for this particular work.